### PR TITLE
49 signup

### PR DESCRIPTION
**Pull request by reemsh2000**
signup component to create a new account
sign

### CODE REVIEW

**Review by NadiaJalalTamimi — APPROVED** [review]

**Review by modrayyes — APPROVED** [review]

File: client/src/components/SignupForm.jsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+          name="basic"
+          labelCol={{ span: 8 }}
+          wrapperCol={{ span: 7 }}
+          initialValues={{ remember: true }}
+          autoComplete="off"
+          onFinish={onFinish}
+          className="signupForm"
+        >
+          <HeaderTitle text="إنشاء حساب" level={2} className="signupTitle" />
+          <Form.Item
```
Comment (Nizar7zak) [inline client/src/components/SignupForm.jsx:48]: you can destruct Item form Form

File: client/src/components/SignupForm.jsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  const [errorMsg, setErrorMsg] = useState('');
+  const errorlert = (erroMsg) => {
+    message.error(erroMsg);
+  };
+  const postData = async (userInfo) => {
+    try {
+      const { msg } = await httpService.post('/api/v1/signup', userInfo);
+      setErrorMsg(msg);
+    } catch (error) {
+      setErrorMsg(error);
```
Comment (Nizar7zak) [inline client/src/components/SignupForm.jsx:23]: here check if the error is expected or not 
if (error.status > 400 && error.status < 500} {
set ErrorMsg...
}

you must use useEffect and stop request in return function.
take a look at service file, to stop request

File: client/src/components/SignupForm.jsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import Img from './common/Img';
+import ParagraphText from './common/Paragraph';
+import HeaderTitle from './common/Title';
+import 'antd/dist/antd.css';
+import './signup.css';
+
+function SignupForm() {
+  const [errorMsg, setErrorMsg] = useState('');
+  const errorlert = (erroMsg) => {
+    message.error(erroMsg);
```
Comment (Nizar7zak) [inline client/src/components/SignupForm.jsx:16]: you can use it directly

File: client/src/components/SignupForm.jsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            name="password"
+            rules={[{ required: true, message: 'كلمة المرور مطلوبة' }]}
+          >
+            <Input.Password placeholder=" كلمة المرور " className="signupInput" />
+          </Form.Item>
+          <Form.Item
+            name="confirmPassword"
+            rules={[{ required: true, message: '  تأكيد كلمة المرور مطلوبة' }]}
+          >
+            <Input.Password placeholder=" تأكيد كلمة المرور " className="signupInput" />
```
Comment (Nizar7zak) [inline client/src/components/SignupForm.jsx:72]: here, the same thing, you can destruct password from input

File: client/src/components/SignupForm.jsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+          </Form.Item>
+          <Form.Item
+            name="originalCity"
+            rules={[{ required: true, message: '  البلدة الأصلية مطلوبة' }]}
+          >
+            <Input placeholder="  البلدة الأصلية   " className="signupInput" />
+          </Form.Item>
+
+          <Form.Item wrapperCol={{ offset: 8, span: 16 }}>
+            <Button type="primary" htmlType="submit" className="signupSubmit signupBtn">
```
Comment (Nizar7zak) [inline client/src/components/SignupForm.jsx:82]: why you didn't use our Button common component

**Review by Nizar7zak — CHANGES_REQUESTED** [review]

File: client/src/components/SignupForm.jsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+              إنشاء حساب
+            </Button>
+            <Button type="primary" htmlType="submit" className="homePageButton signupBtn">
+              الصفحة الرئيسية
+            </Button>
+          </Form.Item>
+          <div className="signupRedirect">
+            <ParagraphText text="هل تمتلك حساب ؟" strong={false} />
+            <Link to="/api/v1/signup" className="loginLink">تسجيل الدخول</Link>
```
Comment (LinaYahya) [inline client/src/components/SignupForm.jsx:91]:
```suggestion
            <Link to="/signup" className="loginLink">تسجيل الدخول</Link>
```

File: client/.env [hunk 7]
```diff
@@ -1,2 +1,3 @@
 ESLINT_NO_DEV_ERRORS=true
```
Comment (LinaYahya) [inline client/.env:1]: delete this file and add to .gitignore

File: client/src/App/index.js [hunk 8]
```diff
@@ -1,6 +1,6 @@
 /* eslint-disable no-console */
```
Comment (LinaYahya) [inline client/src/App/index.js:1]:
```suggestion

```

File: client/src/components/SignupForm.jsx [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+    } catch (error) {
+      setErrorMsg(error);
+    }
+  };
+
+  const onFinish = (values) => {
+    const { confirmPassword, password } = values;
+    if (confirmPassword === password) {
+      postData(values);
+    }
```
Comment (LinaYahya) [inline client/src/components/SignupForm.jsx:31]: what if not the same

**Review by LinaYahya — CHANGES_REQUESTED** [review]

File: client/src/components/SignupForm.jsx (hunk 9 above)
Comment (reemsh2000) [reply client/src/components/SignupForm.jsx:31]: done

File: client/src/App/index.js (hunk 8 above)
Comment (reemsh2000) [reply client/src/App/index.js:1]: done

File: client/.env (hunk 7 above)
Comment (reemsh2000) [reply client/.env]: done

File: client/src/components/SignupForm.jsx (hunk 6 above)
Comment (reemsh2000) [reply client/src/components/SignupForm.jsx:91]: done

File: client/src/components/SignupForm.jsx (hunk 3 above)
Comment (reemsh2000) [reply client/src/components/SignupForm.jsx:16]: done

File: client/src/components/SignupForm.jsx (hunk 4 above)
Comment (reemsh2000) [reply client/src/components/SignupForm.jsx:72]: done

File: client/src/components/SignupForm.jsx (hunk 2 above)
Comment (reemsh2000) [reply client/src/components/SignupForm.jsx:23]: done

File: client/src/components/SignupForm.jsx (hunk 1 above)
Comment (reemsh2000) [reply client/src/components/SignupForm.jsx:48]: done

File: client/src/components/SignupForm.jsx [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+import { Link } from 'react-router-dom';
+import http from '../services/httpService';
+import Img from './common/Img';
+import ParagraphText from './common/Paragraph';
+import HeaderTitle from './common/Title';
+import './signup.css';
+
+const postData = async (userInfo) => {
+  try {
+    await http.post('/api/v1/signup', userInfo);
```
Comment (LinaYahya) [inline client/src/components/SignupForm.jsx:14]: I think we need to setUser to userData, So we'll need to get user data from the server and add setUser to auth provider.
then change this function to receive callbak wich will setUser to user data and then `history.push('/')`

File: client/src/components/SignupForm.jsx [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+  } catch (error) {
+    if (error.status >= 400 && error.status <= 500) {
```
Comment (LinaYahya) [inline client/src/components/SignupForm.jsx:16]: no need to this condition as we handle this in the http

File: client/src/components/SignupForm.jsx [hunk 12]
```diff
@@ -0,0 +1,102 @@
+import React from 'react';
```
Comment (LinaYahya) [inline client/src/components/SignupForm.jsx:1]: create a folder for signup and put your jsx and css file to it

**Review by LinaYahya — CHANGES_REQUESTED** [review]

File: client/src/components/SignupForm.jsx (hunk 12 above)
Comment (reemsh2000) [reply client/src/components/SignupForm.jsx:1]: _done_

**Review by Nizar7zak — APPROVED** [review]

**Pull request merged**